File: src/test/java/com/cxxcxx/zinecraft/api/weapon/action/melee/MeleeAttackActionTest.java
package com.cxxcxx.zinecraft.api.weapon.action.melee;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MeleeAttackActionTest {
  @Test
  void usesSwordFinalAttackDamageWithoutApplyingItsBaseTwice() {
    assertEquals(7.0F, MeleeDamage.resolveActionMeleeDamage(7.0F, 7.0));
    assertEquals(8.05F, MeleeDamage.resolveActionMeleeDamage(7.0F, 8.05), 0.0001F);
  }

  @Test
  void keepsDeclaredDamageWhenAttributeDataIsUnavailable() {
    assertEquals(7.0F, MeleeDamage.resolveActionMeleeDamage(7.0F, null));
    assertEquals(7.0F, MeleeDamage.resolveActionMeleeDamage(7.0F, Double.NaN));
  }
}

File: src/test/kotlin/com/cxxcxx/zinecraft/api/weapon/action/melee/MeleeAttackActionTest.kt
package com.cxxcxx.zinecraft.api.weapon.action.melee

import kotlin.test.Test
import kotlin.test.assertEquals

class MeleeAttackActionTest {
  @Test
  fun `uses the sword final attack damage without applying its base twice`() {
    assertEquals(7.0f, resolveActionMeleeDamage(7.0f, 7.0))
    assertEquals(8.05f, resolveActionMeleeDamage(7.0f, 8.05), absoluteTolerance = 0.0001f)
  }

  @Test
  fun `keeps declared damage when attribute data is unavailable`() {
    assertEquals(7.0f, resolveActionMeleeDamage(7.0f, null))
    assertEquals(7.0f, resolveActionMeleeDamage(7.0f, Double.NaN))
  }
}
